spec/datadog/core/configuration/settings_spec.rb

```diff
@@ -497,39 +497,108 @@
+            .to change { settings.profiling.advanced.experimental_allocation_enabled }
+            .from(false)
+            .to(true)
+        end
+      end
+
+      describe '#experimental_heap_enabled' do
+        subject(:experimental_heap_enabled) { settings.profiling.advanced.experimental_heap_enabled }
+
+        context 'when DD_PROFILING_EXPERIMENTAL_HEAP_ENABLED' do
+          around do |example|
+            ClimateControl.modify('DD_PROFILING_EXPERIMENTAL_HEAP_ENABLED' => environment) do
+              example.run
+            end
+          end
+
+          context 'is not defined' do
+            let(:environment) { nil }
+
+            it { is_expected.to be false }
+          end
+
+          [true, false].each do |value|
+            context "is defined as #{value}" do
+              let(:environment) { value.to_s }
+
+              it { is_expected.to be value }
+            end
           end
         end
       end
 
-      describe '#allocation_counting_enabled=' do
-        it 'updates the #allocation_counting_enabled setting' do
-          settings.profiling.advanced.allocation_counting_enabled = true
+      describe '#experimental_heap_enabled=' do
+        it 'updates the #experimental_heap_enabled setting' do
+          expect { settings.profiling.advanced.experimental_heap_enabled = true }
+            .to change { settings.profiling.advanced.experimental_heap_enabled }
+            .from(false)
+            .to(true)
+        end
+      end
 
-          expect { settings.profiling.advanced.allocation_counting_enabled = false }
-            .to change { settings.profiling.advanced.allocation_counting_enabled }
-            .from(true)
-            .to(false)
+      describe '#experimental_allocation_sample_rate' do
+        subject(:experimental_allocation_sample_rate) { settings.profiling.advanced.experimental_allocation_sample_rate }
+
+        context 'when DD_PROFILING_EXPERIMENTAL_ALLOCATION_SAMPLE_RATE' do
+          around do |example|
+            ClimateControl.modify('DD_PROFILING_EXPERIMENTAL_ALLOCATION_SAMPLE_RATE' => environment) do
+              example.run
+            end
+          end
+
+          context 'is not defined' do
+            let(:environment) { nil }
+
+            it { is_expected.to be 50 }
+          end
+
+          [100, 30.5].each do |value|
+            context "is defined as #{value}" do
+              let(:environment) { value.to_s }
+
+              it { is_expected.to be value.to_i }
+            end
+          end
+        end
+      end
+
+      describe '#experimental_allocation_sample_rate=' do
+        it 'updates the #experimental_allocation_sample_rate setting' do
+          expect { settings.profiling.advanced.experimental_allocation_sample_rate = 100 }
+            .to change { settings.profiling.advanced.experimental_allocation_sample_rate }
+            .from(50)
+            .to(100)
         end
       end
 
```

Comment:
Not for this PR for sure! :arrow_up: These tests are boiler plate-y, but also not that hard to maintain. I've thought about maybe doing a bit of fancy code to reuse (e.g. test all boolean options trivially, but never went for it ;)